girder_worker/plugins/docker/__init__.py

```diff
@@ -1,12 +1,38 @@
+import subprocess
+
+
 def before_run(e):
     import executor
     if e.info['task']['mode'] == 'docker':
         executor.validate_task_outputs(e.info['task_outputs'])
 
 
+def cleanup(e):
+    """
+    Since files written by docker containers are owned by root, we can't
+    clean them up in the worker process since that typically doesn't run
+    as root. So, we run a lightweight container to make the temp dir cleanable.
+    """
+    from .executor import DATA_VOLUME
+    if e.info['task']['mode'] == 'docker' and '_tempdir' in e.info['kwargs']:
+        tmpdir = e.info['kwargs']['_tempdir']
+        cmd = [
+            'docker', 'run', '--rm', '-v', '%s:%s' % (tmpdir, DATA_VOLUME),
+            'busybox', 'chmod', '-R', 'o+rw', DATA_VOLUME
```

Comment:
Should this be `a+rw` rather than `o+rw`?  If the docker container happens to be the same UID or GID as girder_worker and leaves the files unwritable or unreadable, then wouldn't we fail to remove them otherwise?